src/python/arcor2/cached.py

```diff
@@ -431,6 +437,58 @@ def constant(self, constant_id: str) -> cmn.ProjectConstant:
         except KeyError:
             raise CachedProjectException("Constant not found.")
 
+    def get_by_id(
+        self, obj_id: str
+    ) -> Union[cmn.SceneObject, cmn.BareActionPoint, cmn.NamedOrientation, cmn.ProjectRobotJoints, cmn.Action]:
+
+        if obj_id in self._action_points:  # AP
+            return self._action_points[obj_id]
+        elif obj_id in self._joints.data:  # Joints
+            return self._joints.data[obj_id]
+        elif obj_id in self._orientations.data:  # Orientation
+            return self._orientations.data[obj_id]
+        elif obj_id in self._actions.data:  # Action
+            return self._actions.data[obj_id]
+
+        from arcor2_arserver.globals import LOCK
+
+        if LOCK.scene:
+            if obj_id in LOCK.scene.object_ids:
+                return LOCK.scene.object(obj_id)
+
+        raise CachedProjectException("Object not found.")
+
+    def get_parent_id(self, obj_id: str) -> Optional[str]:
+
+        if obj_id in self._action_points:  # AP
+            return self._action_points[obj_id].parent if self._action_points[obj_id].parent else None
+        elif obj_id in self._joints.data:  # Joints
+            return self._joints.parent[obj_id].id
+        elif obj_id in self._orientations.data:  # Orientation
+            return self._orientations.parent[obj_id].id
+        elif obj_id in self._actions.data:  # Action
+            return self._actions.parent[obj_id].id
+
+        return None
```

Comment:
Pokud je to neznámé IDčko, nemělo by to spíš vyhodit exception?

Reply:
Já jsem to tam původně měl, ale nějak mi s tím nefungoval zámek, tak jsem to pak měnil. Ale je pravda, že jsem od té doby opravil nějaké bugy, tak to můžu ještě vyzkoušet